src/h5web/visualizations/index.tsx

```diff
@@ -112,36 +116,38 @@ export const VIS_DEFS: Record<Vis, VisDef> = {
     Container: NxSpectrumContainer,
     supportsEntity: (entity, metadata) => {
       const group = getNxDataGroup(entity, metadata);
-
       if (!group) {
-        return false;
+        return false; // entity is not a group and doesn't have a `default` attribute
       }
 
       const signal = getAttributeValue(group, 'signal');
-      if (typeof signal !== 'string') {
-        return false;
-      }
+      assertDefined(signal, "Expected 'signal' attribute");
+      assertStr(signal, "Expected 'signal' attribute to be a string");
 
       const dataset = getLinkedEntity(signal, group, metadata);
-      if (
-        !dataset ||
-        !isDataset(dataset) ||
-        !isNumericType(dataset.type) ||
-        !isSimpleShape(dataset.shape)
-      ) {
-        return false;
-      }
+      assertDefined(dataset, `Expected "${signal}" signal entity to exist`);
+      assertDataset(dataset, `Expected "${signal}" signal to be a dataset`);
+
+      assertNumericType(
+        dataset.type,
+        `Expected "${signal}" dataset to have numeric type`
+      );
+      assertSimpleShape(
+        dataset.shape,
+        `Expected "${signal}" dataset to have simple shape`
+      );
 
       const dimsCount = dataset.shape.dims.length;
       if (dimsCount === 0) {
-        return false;
+        throw new Error('Expected dataset to have at least one dimension');
       }
 
       const interpretation = getAttributeValue(dataset, 'interpretation');
-      return (
-        (!isNxInterpretation(interpretation) && dimsCount === 1) || // NxImage already supports datasets with 2+ dimensions
-        interpretation === NxInterpretation.Spectrum
-      );
+      if (isNxInterpretation(interpretation)) {
+        return interpretation === NxInterpretation.Spectrum;
+      }
+
+      return dimsCount === 1; // NxImage already supports datasets with 2+ dimensions
```

Comment:
I don't like this coupling between the two nexus `supportsEntity` functions. I'm thinking of refactoring this to do something similar to `RawVis`: if the two nexus are supported, then `getSupportedVis` should return only one.